subsys/bluetooth/shell/bt.c

```diff
@@ -361,11 +361,13 @@ bool passes_scan_filter(const struct bt_le_scan_recv_info *info, const struct ne
 
 	if (scan_filter.addr_set) {
 		char le_addr[BT_ADDR_LE_STR_LEN] = {0};
-		int err;
+		int ret;
 
-		err = bt_addr_le_to_str(info->addr, le_addr, sizeof(le_addr));
-		if (err != 0) {
-			shell_error(ctx_shell, "Failed to convert addr to string: %d", err);
+		ret = bt_addr_le_to_str(info->addr, le_addr, sizeof(le_addr));
+		if (ret >= sizeof(le_addr)) {
+			shell_error(ctx_shell,
+					"Insufficient buffer to convert addr to string (%d >= %d)",
+					ret, sizeof(le_addr));
```

Comment:
Should this really be a runtime error though, as the size of the `le_addr` buffer if fixed at build-time? Seems like at most we want an assert here, but perhaps even better would be to just use a `(void)` cast for the return value?

Reply:
I was thinking of removing the check altogether - like all the other places, where the function is used.  Better? (in reality, as long as the buffer is large enough - as it is - we don't need any check).

Reply:
nobody is using the return value it seems + I'll just change it to do the same